### ISSUE TITLE
Hunger damages armor

### ISSUE THREAD

**Issue by cessna151** [stujones11/minetest-3d_armor#119]
From a logical standpoint it doesn't make sense. But i understand MT has it's limits. If i should have posted this on the hunger repo, sorry in advance!
:large_orange_diamond:

**Comment by stujones11** [comment]
I agree it's not great, however, the core does not give any indication of what has damaged the player. Ideally I would like to handle all armor damage on punch but not all mods use the proper api to apply damage and instead set the player HP directly :(

There is an issue somewhere for including a source of damage but until that is implemented, the best the armor mod could do is expose `last_punch_time` though the api, that way other mods would be able to reset that for a given player, causing the HP damage to be ignored.

**Comment by cessna151** [comment]
Thanks for your response. This is pretty much what i figured. Just thought i'd ask to be sure :)
:large_orange_diamond:

**Comment by stujones11** [comment]
As of https://github.com/minetest/minetest/commit/dfc81983491417c5cd1c99d7db05e421c409379d it may be possible to avoid this problem in future versions.

**Comment by orbea** [comment]
@stujones11 Any news on this issue? This also affects hunger with hunger_ng.

**Comment by orbea** [comment]
Seems it was not hard to fix.

https://github.com/minetest-mods/3d_armor/pull/6
